### PR TITLE
WIP: add latex custom extension example

### LINKED ISSUE #6031: Latex axis and title labels
It would be nice to improve the aesthetic quality of the figures generated by bokeh, by including the option to add Latex axis labels. The cosmetic aspect is one issue, but in case of complex variables this could affect readability as well if one is forced to use simple ASCII.

There already seems to be some partial support for latex labels introduced in 0.12.2:

- https://bokeh.github.io/blog/2016/9/6/release-0-12-2/
- http://bokeh.pydata.org/en/latest/docs/user_guide/extensions_gallery/latex.html
    bokeh/bokeh#647

Although to me it seems that with this feature you can only place labels on top of the canvas, but not modify the axis labels or title.
Another concerned user:
- http://stackoverflow.com/questions/41664768/use-latex-in-bokeh-axis-label

In summary adding latex support would highly improve the readability, aesthetic quality and the credibility of figures generated by bokeh.

### PR DESCRIPTION
**Pull request by IuryPiva**
Added an example on how to use MathJax to render latex labels

- [x] issues: related to #6031
- [x] tests added / passed
- [x] release document entry (if new feature or API change)

### CODE REVIEW

**Comment by mattpap** [comment]
You need to rebase this on top of `origin/branch-2.4` and reset your local `branch-2.4`:
```
git rebase --onto origin/branch-2.4 HEAD~1
```

File: examples/models/file/latex_extension_with_mathjax.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    """)
+    height = Nullable(Float, help="""
+    The gradient of the line, in data units
+    """)
+    """A subclass of `Label` with all of the same class attributes except
+    canvas mode isn't supported and DOM manipulation happens in the TypeScript
+    superclass implementation that requires setting `render_mode='css'`).
+
+    Only the render method of LabelView is overwritten to perform the
+    text -> latex (via katex) conversion
```
Comment (tcmetzger) [inline examples/models/file/latex_extension_with_mathjax.py:27]: Shouldn't this be "Mathjax" here?
```suggestion
    text -> latex (via MathJax) conversion
```

**Comment by tcmetzger** [comment]
I just committed an update to the docs as well. This is basically just adding a link to the [existing docs page](https://docs.bokeh.org/en/latest/docs/user_guide/extensions_gallery/latex.html) that has a KaTeX example on it. This is the page that shows up when people search for "LaTeX" in the docs, so I think it is helpful to mention the MathJax example there as well.

Looking through the repo, I found several different versions of the KaTeX example:

* `latex_extension_base` in `examples/custom`
* `latex_extension_full` in `examples/custom`
* `latex_extension.py` in `examples/models/file`
* `extensions_example_latex.py` in `sphinx\source\docs\user_guide\examples` (this is the one that is embedded in the [user guide's extensions gallery](http://localhost:5009/en/latest/docs/user_guide/extensions_gallery/latex.html))

At first glance, these four KaTeX examples seem somewhat similar. @bryevdv @mattpap Is there a quick explanation why there are four examples in different locations? And where should the MathJax example be located (currently in `examples/models/file`)?

**Comment by bryevdv** [comment]
@tcmetzger the `models/file` one was the original simple one. The one under `sphinx` just exists to get it in the docs page (I think it has some accommodations to afford better formatting inside the docs). The other two examples are ones that @mattpap added to demonstrate different methods of building extensions more than anything else (i.e. more than being demonstrations of latex rendering).

The current situation is unfortunate because we "burned" the name "latex" so early on the original katex example. We could simple rename things to have a consistent scheme, e.g

* `latex_extension.py` -> `latex_extension_katex.py`
* add `latex_extension_mathjax.py`
* rename `latex_extension_base` and `latex_extension_full` to `wrapped_extension_full`, etc. 

Doing this might break some links that are out in the wild but I'm personally OK with that. Alternatively we can keep all the current names and just live with `latex_extension_mathjax.py` being kind of inconsistent. I am fine with whatever you decide is best.

**Comment by tcmetzger** [comment]
As much as I'd like to never break any links, I'm leaning more towards renaming as you suggested. With some very cursory googling, I only found a small handful of external links, and most of them point to the documentation page, which will stay the same. Also, especially once we have more LaTeX-related files and documentation, it would be helpful not to have the existing KaTex-based LaTeX things getting confused with the MathJax-based LaTeX integration we are working on...

**Comment by tcmetzger** [comment]
@bryevdv I renamed and moved the files as suggested. This PR is to branch2-4 but I think it could also be ported to 2.3. What do you think?

**Comment by bryevdv** [comment]
@tcmetzger I would prefer to land all this new latex work on a new release branch.

**Comment by mattpap** [comment]
This still needs a rebase.

File: .github/workflows/bokeh-ci.yml [hunk 2]
```diff
@@ -441,6 +441,10 @@ jobs:
         shell: bash -l {0}
         run: bash ci/install_downstream_packages.sh
 
+      - name: Install downstream packages
+        shell: bash -l {0}
+        run: bash ci/install_downstream_packages.sh
+
```
Comment (bryevdv) [inline .github/workflows/bokeh-ci.yml:447]: This change should be reverted.

File: bokeh/core/_templates/autoload_js.js [hunk 3]
```diff
@@ -91,6 +91,8 @@ calls it with the rendered model.
       document.body.appendChild(element);
     }
 
+    const hashes = {{ bundle.hashes|json }};
```
Comment (bryevdv) [inline bokeh/core/_templates/autoload_js.js:94]: I think this PR needs a rebase on current `branch-2.4`? These changes are restoring codepaths that were intentionally removed in a previous PR.

Comment (IuryPiva) [reply bokeh/core/_templates/autoload_js.js:94]: Whenever I pulled after rebasing these 2 files would update. I tried rebasing and then force pushing now

Comment (IuryPiva) [reply bokeh/core/_templates/autoload_js.js:94]: ❯ git remote get-url upstream
https://github.com/bokeh/bokeh.git

❯ git rebase --onto upstream/branch-2.4 HEAD~1
`

Comment (bryevdv) [reply bokeh/core/_templates/autoload_js.js:94]: @IuryPiva did you fetch the most recent `branch-2.4` before the rebase? If not, you need to do that first. I would actually expect a force push is necessary after the rebase.

Comment (IuryPiva) [reply bokeh/core/_templates/autoload_js.js:94]: I did, and it worked on [11060](https://github.com/bokeh/bokeh/pull/11060). I will try again

Comment (bryevdv) [reply bokeh/core/_templates/autoload_js.js:94]: I'm not sure what's going on, it actually added a third copy of that block. @tcmetzger is going to add you to @bokeh/dev so that you can work off a clone of the main repo instead of a fork. I think it might be simplest just to submit new PRs after that is done.

Comment (bryevdv) [reply bokeh/core/_templates/autoload_js.js:94]: @Iurypiva you should have an invite to @bokeh/dev now

**Comment by IuryPiva** [comment]
Superseded https://github.com/bokeh/bokeh/pull/11067

**Comment by bryevdv** [comment]
superseded by #11067

**Comment by bryevdv** [comment]
Weird it was not showing the existing superseded comment and tag, this it did 🤷
